vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddBiblioBlockController.java

```diff
@@ -0,0 +1,51 @@
+package edu.asu.diging.vspace.web.staff;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.asu.diging.vspace.core.model.IBiblioBlock;
+import edu.asu.diging.vspace.core.model.impl.BiblioBlock;
+import edu.asu.diging.vspace.core.services.IContentBlockManager;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.web.bind.annotation.*;
+
+
+@Controller
+public class AddBiblioBlockController {
+
+    @Autowired
+    private IContentBlockManager contentBlockManager;
+
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/bibliography", method = RequestMethod.POST)
+    public ResponseEntity<String> addBiblioBlock(@PathVariable("id") String slideId,
+            @PathVariable("moduleId") String moduleId, @RequestBody String biblioBlockData) throws JsonProcessingException {
```

Comment:
Can't you have spring parse the json into an object (just create a BiblioBlockData class with the necessary fields and use it here)? This way you can drop line 35-41.